public/app/features/variables/adhoc/picker/AdHocFilter.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { PureComponent, ReactNode } from 'react';
+import { AdHocVariableFilter } from 'app/features/variables/types';
+import { DataSourceRef, SelectableValue } from '@grafana/data';
+import { AdHocFilterBuilder } from './AdHocFilterBuilder';
+import { ConditionSegment } from './ConditionSegment';
+import { REMOVE_FILTER_KEY } from './AdHocFilterKey';
+import { AdHocFilterRenderer } from './AdHocFilterRenderer';
+
+interface Props {
+  datasource: DataSourceRef | null;
+  filters: AdHocVariableFilter[];
+  addFilter: (filter: AdHocVariableFilter) => void;
+  removeFilter: (index: number) => void;
+  changeFilter: (index: number, newFilter: AdHocVariableFilter) => void;
+}
+
+/**
+ * Simple filtering component that automatically uses datasource APIs to get available labels and it's values, for
+ * dynamic visual filtering without need for much setup. Instead of having single onChange prop this reports all the
+ * change events with separate props so it is usable with AdHocPicker.
+ *
+ * Note: There isn't API on datasource to suggest the operators here so that is hardcoded to use prometheus style
+ * operators. Also filters are assumed to be joined with `AND` operator, which is also hardcoded.
+ */
+export class AdHocFilter extends PureComponent<Props> {
```

Comment:
@hugohaggmark @grafana/user-essentials I created a separate component for this so it is easier to use outside of the dashboard variable domain. This means there is:
`AdHocFilter`
Simple-ish component that just takes a datasourceRef and based on that uses the DatasourceAPI to create a simple filtering UI.
`AdHocPicker`:
The same thing but also connects to redux and expects variable model so it ties to a notion of dashboard ad hoc variable.

Let me know if this makes sense and whether this can stay here, maybe in the future it may go into grafana-ui but also it may be a bit too tied to the datasourceAPI so not sure if grafana-ui is the best fit in it's current shape.

Reply:
Gotcha, I think it's ok to separate them like this. But the current implementation breaks the variable case



https://user-images.githubusercontent.com/562238/139790115-c4611168-0bb6-49d7-8cac-24f4694fdff2.mp4

.

Reply:
Ah thanks, yeah did not test it much, wanted to check if this makes sense in general. Will also add some tests.

Reply:
Should be fixed now